src/main/java/edu/harvard/iq/dataverse/util/MailUtil.java

```diff
@@ -32,6 +32,10 @@ public static InternetAddress parseSystemAddress(String systemEmail) {
 
     public static String getSubjectTextBasedOnNotification(UserNotification userNotification, Object objectOfNotification) {
         List<String> rootDvNameAsList = Arrays.asList(BrandingUtil.getInstallationBrandName());
+        String datasetDisplayName = "";
+        if (objectOfNotification != null) {
+            datasetDisplayName = ((Dataset)objectOfNotification).getDisplayName();
```

Comment:
Isn't this method called with non-Dataset objectOfNotifcation? which would cause a class cast exception?

Reply:
@qqmyers - I will check this.

Reply:
@qqmyers : I didn't come across any exception while building the package.

Reply:
It wouldn't be a build time issue. If you look at MailUtilServiceBean.getObjectOfNotification(), it can return a Dataset, Datafile, DatasetVersion, etc. I think when you change is called for one of the message types that doesn't have a Dataset objectOfNotification, you'd get a runtime class cast exception. I think just adding `&& objectOfNotification instanceof Dataset` would avoid the problem.

Reply:
@qqmyers  - updated the program